[bug] Removed automatic clamping for unbounded spectrum

DawyD opened the pull request:
paramaters_changed no longer clamps RGB/monochromatic values if unbounded property was specified

## Description

When changing the unbounded RGB/monochromatic spectrum through parameters update in python, the values were clamped to [0,1]. This was obviously unwanted as it e.g. prevented from changing eta/k parameters of the conductor BSDF to realistic values.

I have added a member variable m_unbounded to the SRGBReflectanceSpectrum which is initialized based on the properties and when python parameter update invoke the parameters_changed, the clamping now occurs only if the spectrum is unbounded.

## Testing

*Please describe the tests that you added to verify your changes.*

## Checklist:

*Please make sure to complete this checklist before requesting a review.*

- [X] My code follows the [style guidelines](https://mitsuba2.readthedocs.io/en/latest/src/developer_guide/intro.html#introduction) of this project
- [X] My changes generate no new warnings
- [ ] My code also compiles for `gpu_*` and `packet_*` variants. If you can't test this, please leave below
- [ ] I have commented my code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I cleaned the commit history and removed any "Merge" commits
- [X] I give permission that the Mitsuba 2 project may redistribute my contributions under the terms of its [license](https://github.com/mitsuba-renderer/mitsuba2/blob/master/LICENSE)

Speierers commented:
Hi @DawyD ,

Thanks for this PR, this change looks good to me. @wjakob any objections?

Speierers commented:
@DawyD before we merge this it would be great if you could add a discription of the `unbounded` parameter to the docstring at the top of this plugin so that it shows up in the documentation page 📖

DawyD commented:
Hi @Speierers ,
I've added a very basic description of the parameters. The color property is of type ScalarColor3f therefore, I've added a color paramtype to the documentation configuration. 

I am unaware of the difference between the [conf.py](https://github.com/mitsuba-renderer/mitsuba2/blob/master/docs/conf.py) and [docs_api/conf.py](https://github.com/mitsuba-renderer/mitsuba2/blob/master/docs/docs_api/conf.py), so I've updated both of them. Moreover, docs_api/conf.py used `add_stylesheet` method which is deprecated since Sphinx 1.8.0 so I changed it to `add_css_file` as in conf.py to test the compilation.
